src/draw/ElkLayout.js

```diff
@@ -0,0 +1,377 @@
+ * Then we can compute the size of parents at this level and call ELK at the level above.
+ * We do this at each depth, starting from the deepest.
+ *
+ * {DefaultData} has a flat data structure for its components. Parents and children are
+ * stored at the same level, and the parent-child relation is stored in attributes.
+ *
+ * To separate the differents levels of hierarchy in the graph, we need to build a tree.
+ * In this file, we call components the units coming from the Leto plugin, and nodes their
+ * reference in the tree (with additional information to match ELK format later on).
+ * A component always has exactly one matching node.
+ * Therefore {ElkLayout.getNodes} takes the components (flat structure) as an argument and
+ * return the nodes (tree structure).
+ * @see {ElkLayout.getNodes}
+ * @description
+ *
+ * We only ever call ELK at a single level of hierarchy, so we cannot have cross-depth/level links.
+ * We represent a link between two nodes from different depth in the tree by a link  between their
+ * respective ancestors (that we get with {ElkLayout.getAncestorByDepth} at a given level.
+ */
+
+/**
+ * A component data for automatic layout.
+ * Used within {@link ElkLayout}.
+ *
+ * It has tree-related fields : parent, children, depth,
+ * in addition to the fields from {@link Component} in {@link raw}.
+ * @typedef {object} NodeData
+ * @property {NodeData} [parent] - The node's parent.
+ * @property {Component} [raw] - The underlying component.
+ * @property {NodeData[]} children - This node's children.
+ * @property {number} depth - distance from the root (root's depth is 0).
+ */
+
+/**
+ * An ELK node.
+ * (external)
+ * @typedef {object} ElkNode
+ * @property {string} id - id.
+ * @property {ElkNode[]} [children] - children.
+ * @property {ElkPort[]} [ports] - ports.
+ * @property {ElkExtendedEdge[]} [edges] - edges.
+ * @see https://eclipse.dev/elk/documentation/tooldevelopers/graphdatastructure/jsonformat.html
+ * @example
+ *
+ * // from Typescript
+ * export interface ElkNode extends ElkShape {
+ *     id: string
+ *     children?: ElkNode[]
+ *     ports?: ElkPort[]
+ *     edges?: ElkExtendedEdge[]
+ * }
+ */
+
+/**
+ * @augments DefaultLayout
+ * Various tools to arrange the graph view (semi-)automatically.
+ *
+ * Uses among other things Eclipse Layout Kernel (aka. ELK).
+ *
+ * ELK's documentation is available at:
+ * {@link https://eclipse.dev/elk/}
+ *
+ * What we use here elkjs, which is a transpiled version
+ * from Java to Javascript :
+ * {@link https://github.com/kieler/elkjs}
+ */
+class ElkLayout extends DefaultLayout {
+  /**
+   * ELK import.
+   * @type {ELK}
+   * @private
+   * @constant
+   */
+  static elk = new ELK();
+
+  /**
+   * Initializes ELK parameters and inherited fields.
+   * @param {DefaultData} pluginData - a graph to be arranged.
+   * @param {object} [elkParams] - Parameters for the layout algorithm. (use defaults if unsure)
```

Comment:
add @link to elk parameters documentation.

Reply:
done